Scripts/RemediationScripts/Remediate-ConfigureWAFPolicyForFrontDoor.ps1

```diff
@@ -0,0 +1,604 @@
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    
+ 
+        Write-Host "[Step 5 of 5] Configuring WAF Policy for Front Door Endpoint(s)"
+        Write-Host $([Constants]::SingleDashLine)
+
+        # To hold results from the remediation.
+        $frontDoorFrontendpointsRemediated = @()
+ 
+        # Remidiate Controls by Configuring WAF Policy
+        $frontDoorEndpointsWithWAFPolicyNotConfigured | ForEach-Object {
+            $frontDoorEndPoint = $_
+            $frontdoorName = $_.FrontDoorName;
+            $resourceGroupName = $_.ResourceGroupName; 
+
+            # Holds the list of Front Door Endpoints without WAF Policy Configure
+            $frontendpointsSkipped = @()
+            $frontendpointsSkippedStr = [String]::Empty
+             
+            try
+            {   
+                Do
+                {
+                    Write-Host "* Press S to skip this Front Door & move to next Front Door" 
+                    Write-Host $([Constants]::SingleDashLine)
+                    $wafPolicyName = Read-Host -Prompt "Enter WAF Policy Name for Endpoint: [$($_.EndPointName)] of Frontdoor [$($frontdoorName)] " 
+                    if ( $wafPolicyName -eq "S" ) { return }
+                    Write-Host $([Constants]::SingleDashLine)
+                    Write-Host "* Press S to skip this Front Door & move to next Front Door" 
+                    Write-Host $([Constants]::SingleDashLine)
+                    $policyResourceGroup = Read-Host -Prompt "Enter WAF Policy Resource Group Name for Endpoint: [$($_.EndPointName)] of Frontdoor [$($frontdoorName)] "
+                    if ( $policyResourceGroup -eq "S" ) { return }
+
+                    $wafPolicyName = $wafPolicyName.Trim()
+                    $policyResourceGroup = $policyResourceGroup.Trim()
+
+                    Write-Host $([Constants]::SingleDashLine)
+                    $policy = Get-AzFrontDoorWafPolicy -Name  $wafPolicyName  -ResourceGroupName $policyResourceGroup -ErrorAction SilentlyContinue
+
+                    if($policy -eq $null)
+                    {
+                        Write-Host "WAF Policy name or WAF Policy Resource Group Name is not correct. Please enter correct details."
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+
+                    if($policy -ne $null -and $policy.Sku -ne 'Classic_AzureFrontDoor')
+                    {
+                        Write-Host "WAF Policy is not of type Front door tier Classic . Please enter correct details."
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+                while($policy.Sku -ne 'Classic_AzureFrontDoor' -or $policy -eq $null)
+
+                Write-Host "Enabling Prevention Mode on provided WAF Policy" -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
please remove the Prevention mode setting, we will only configure the policy in this BRS

Reply:
removed